### PR TITLE
New metric: Davies bouldin score

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes part of #2003 

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
:books: Documentation preview :books:: https://torchmetrics--2071.org.readthedocs.build/en/2071/

### CODE REVIEW

File: src/torchmetrics/clustering/calinski_harabasz_score.py [hunk 1]
```diff
@@ -106,20 +106,20 @@ def plot(self, val: Union[Tensor, Sequence[Tensor], None] = None, ax: Optional[_
 
             >>> # Example plotting a single value
             >>> import torch
-            >>> from torchmetrics.clustering import RandScore
-            >>> metric = RandScore()
-            >>> metric.update(torch.randint(0, 4, (10,)), torch.randint(0, 4, (10,)))
+            >>> from torchmetrics.clustering import CalinskiHarabaszScore
```
Comment (Borda) [inline src/torchmetrics/clustering/calinski_harabasz_score.py:109]: nice catch :)

**Review by Borda — APPROVED** [review]

**Review by Borda — COMMENTED** [review]
btw, seems you are missing docs

**Review by justusschock — APPROVED** [review]
